crates/libcontainer/src/workload/mod.rs

```diff
@@ -0,0 +1,77 @@
+use anyhow::{Context, Result};
+use oci_spec::runtime::Spec;
+
+use self::default::DefaultExecutor;
+#[cfg(feature = "wasm-wasmer")]
+use self::wasmer::WasmerExecutor;
+
+pub mod default;
+#[cfg(feature = "wasm-wasmer")]
+pub mod wasmer;
+
+static EMPTY: Vec<String> = Vec::new();
+
+pub trait Executor {
+    /// Executes the workload
+    fn exec(&self, spec: &Spec) -> Result<()>;
```

Comment:
Do any of the functions for Executor actually need a reference to `self`? The CompositeExecutor is the only one that uses it, and it probably doesn't even need to satisfy this trait. If none of the Executors needed a `&self`, you'd also avoid the need to ever instantiate Executors, which aside from the CompositeExecutor are all empty structs.